### ISSUE TITLE
error setting option in graph2d

### ISSUE THREAD

**Issue by gmlevin** [almende/vis#2458]
I want to auto scale my y-axis in graph2d. (I also want to be able to switch between fixed and autoscaled limits.) I tried setting dataAxis.left.range.max = null or undefined. Passing the value null (not the string) seems to get the desired behavior, but reports the following error. Passing undefined has the same issue.
I see this behavior with vis-4.17.0 and previous versions.

I am assuming it is just the max field that is having a problem and that it is being set to undefined, which is what I want, and other fields are being set correctly. This appears to be true, because when I set min to undefined, both settings are reported as errors and they both are autoscaled.
```
Invalid type received for "max". Expected: number. Received [null] "null"
Problem value found at: 
options = {
  dataAxis: {
    left: {
      range: {
        max
      }
    }
  }
}
vis.js:26447 Errors have been found in the supplied options object.
```

**Comment by Tooa** [comment]
Can you provide a jsbin showing the issue and describe the steps to reproduce it? Then, I will be able to debug the code and find the problem.

**Comment by gmlevin** [comment]
I have created a jsbin based on the "Graph2d | Dynamic Styling Example".
How do I make it available to you? I exported it as a gist to
https://gist.github.com/anonymous/4d0e6235e422c906d7a427f60c79041b.

Setting the top option will set the maximum on the y-axis to 50 (larger
than needed), 10 (smaller than the actual max), undefined or null to get
the dynamic max and see the error on the console.

Hope this helps.

Gary Levin

**Comment by Tooa** [comment]
I can reproduce this issue with the jsfiddle provided [here](https://jsbin.com/gifodew/edit?html,output). Just set `set max` to `undefined` and the error appears in the console. Since the [documentation](http://visjs.org/docs/graph2d/) states `All options are optional.` I guess `undefined` should be a valid input. What do you think @mojoaxel ?

Not passing the `max` field results in the expected behavior. So, as a temporal solution you can use something like [this](http://underscorejs.org/#omit) (function omit) to remove the max field from the options value instead of overwriting it with `undefined`.

**Comment by gmlevin** [comment]
What are you applying "omit" to? I believe that set_options does an update,
so if I omit the option from the argument, it stays what it had been. I
need to explicitly set it to undefined.  If you can apply omit to the
stored options, that would do what I need. Otherwise, I'll just let it
generate the error message.

Thanks for the pointer to the library. I expect it will be useful in the
future, even if it does not solve this problem.

Gary

**Comment by Tooa** [comment]
My bad. You are right. I found the issue and fixed it in https://github.com/almende/vis/pull/2634. We will release a new version including that fix this weekend. Thanks for reporting the issue.

**Comment by gmlevin** [comment]
Thanks for the fix. VisJS is a great package and I like using it for many
displays.

**Comment by yotamberk** [comment]
closed with #2634 . thanks @Tooa !
